tests/st/calicoctl/test_crud.py

```diff
@@ -624,6 +624,55 @@ def test_metadata_unchanged(self, create_cmd, update_cmd):
         # Delete the resource without using a resource version.
         rc = calicoctl("delete hostendpoint %s" % name(rev1))
         rc.assert_no_error()
+
+    def test_export_flag(self):
+        """
+        Test that the cluster-specific information gets stripped
+        out of a "get" request.
+        """
+        # Create a new Network Policy with all metadata specified
+        rc = calicoctl('create', data=networkpolicy_name2_rev1)
+        rc.assert_no_error()
+        rc = calicoctl(
+            "get networkpolicy %s -o yaml" % name(networkpolicy_name2_rev1))
+        rc.assert_no_error()
+        rev0 = rc.decoded
+        self.assertIn('uid', rev0['metadata'])
+        self.assertIn('creationTimestamp', rev0['metadata'])
+        self.assertEqual(rev0['metadata']['name'], networkpolicy_name2_rev1['metadata']['name'])
+        self.assertEqual(rev0['metadata']['namespace'], networkpolicy_name2_rev1['metadata']['namespace'])
+        self.assertIn('resourceVersion', rev0['metadata'])
+
+        # Retrieve the Network Policy with the export flag and
+        # verify that cluster-specific information is not present
+        rc = calicoctl(
+            "get networkpolicy %s -o yaml --export" % name(networkpolicy_name2_rev1))
```

Comment:
does this work with the list operation as well? (when you don't specify the resource name), looks like k8s doesn't do anything when `--export` is specified on a list (get without a resource name/namespace)

Reply:
Good point, this would currently work with the list operation (at least based off my understanding of the code) and it probably shouldn't to match k8s.

Reply:
I changed this functionality to silently ignore the `--export` flag if a name is not provided. From my testing, this should mirror k8s.